app/controllers/wizard/steps/base_controller.rb

```diff
@@ -3,15 +3,23 @@ module Steps
     class BaseController < ::ApplicationController
       default_form_builder GOVUKDesignSystemFormBuilder::FormBuilder
 
-      before_action :assign_commodity
+      helper_method :commodity
 
-      def assign_commodity
-        @commodity = Api::Commodity.build(
+      def commodity
+        @commodity ||= Api::Commodity.build(
           service_choice,
           commodity_code,
         )
       end
 
+      def xi_commodity_for(geographical_area_id)
```

Comment:
Isn't this only used for the origin step? I think it's cleaner to move it out to the one place it's used?

Reply:
I think it will be shared by other routes as well.